subsys/dfu/boot/mcuboot_shell.c

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2020 Grinn
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <dfu/mcuboot.h>
+#include <init.h>
+#include <shell/shell.h>
+#include <stdlib.h>
+
+#include "mcuboot_priv.h"
+
+struct area_desc {
+	const char *name;
+	unsigned int id;
+};
+
+static const struct area_desc areas[] = {
+	{"primary", FLASH_AREA_IMAGE_PRIMARY},
+	{"secondary", FLASH_AREA_IMAGE_SECONDARY},
+};
+
+static const char *swap_state_magic_str(uint8_t magic)
+{
+	switch (magic) {
+	case BOOT_MAGIC_GOOD:
+		return "good";
+	case BOOT_MAGIC_BAD:
+		return "bad";
+	case BOOT_MAGIC_UNSET:
+		return "unset";
+	case BOOT_MAGIC_ANY:
+		return "any";
+	case BOOT_MAGIC_NOTGOOD:
+		return "notgood";
+	}
+
+	return "unknown";
+}
+
+static const char *swap_type_str(uint8_t type)
+{
+	switch (type) {
+	case BOOT_SWAP_TYPE_NONE:
+		return "none";
+	case BOOT_SWAP_TYPE_TEST:
+		return "test";
+	case BOOT_SWAP_TYPE_PERM:
+		return "perm";
+	case BOOT_SWAP_TYPE_REVERT:
+		return "revert";
+	case BOOT_SWAP_TYPE_FAIL:
+		return "fail";
+	}
+
+	return "unknown";
+}
+
+static const char *swap_state_flag_str(uint8_t flag)
+{
+	switch (flag) {
+	case BOOT_FLAG_SET:
+		return "set";
+	case BOOT_FLAG_BAD:
+		return "bad";
+	case BOOT_FLAG_UNSET:
+		return "unset";
+	case BOOT_FLAG_ANY:
+		return "any";
+	}
+
+	return "unknown";
+}
+
+static int cmd_mcuboot_erase(const struct shell *shell, size_t argc,
+			     char **argv)
+{
+	unsigned int id;
+	int err;
+
+	id = strtoul(argv[1], NULL, 0);
+
+	err = boot_erase_img_bank(id);
```

Comment:
exacts area_id is defined in compile time,
User might erase any area which match given `ID`, not only these `IDs` which belongs to image banks.

Reply:
This command should limit `ID` used somehow.

Reply:
I wonder if we should limit `ID` in `mcuboot_shell.c` implementation or rather inside `boot_erase_img_bank()` body. In general I share your concerns here, about (not) giving user possibility to do stupid stuff over shell. However I have the same concern when directly using `boot_erase_img_bank()`, which is needed when something other than mcumgr is supposed to be used for DFU. I've written `mcuboot_shell.c` to basically get developers the same access as using C API directly, but in a much easier way, hence using shell interface.

So in summary I lean towards adding a check in `boot_erase_img_bank()`.

@nvlsianpu @utzig please share your opinion o that.

Reply:
> So in summary I lean towards adding a check in `boot_erase_img_bank()`.

Yes, this would be right approach IMO.

Reply:
I've created PR #30060, where we can discuss how to restrict `boot_erase_img_bank()`. I will leave this PR as is.